base/src/main/java/org/aya/core/serde/SerDef.java

```diff
@@ -30,8 +32,9 @@ record Fn(
     @NotNull SerTerm result
   ) implements SerDef {
     @Override public @NotNull Def de(SerTerm.@NotNull DeState state) {
+      DefVar<FnDef, Decl.FnDecl> ref = state.def(name);
       return new FnDef(
-        state.def(name), telescope.map(tele -> tele.de(state)),
+        ref, telescope.map(tele -> tele.de(state)),
```

Comment:
Why this change?

Reply:
I am not sure whether the arguments are evaluated from left to right or right to left or unspecified.

Reply:
Does it matter?

Reply:
I tried reverting these changes and nothing was broken. Thanks for pointing them out!